mslib/mscolab/server.py

```diff
@@ -351,6 +352,40 @@ def get_user():
     return json.dumps({'user': {'id': g.user.id, 'username': g.user.username}})
 
 
+@APP.route('/upload_profile_image', methods=["POST"])
+@verify_user
+def upload_profile_image():
+    user_id = request.form['user_id']
+    file = request.files['image']
+    if not file:
+        return jsonify({'message': 'No file provided or invalid file type'}), 400
+    if not file.mimetype.startswith('image/'):
+        return jsonify({'message': 'Invalid file type'}), 400
+    if file.content_length > mscolab_settings.MAX_UPLOAD_SIZE:
+        return jsonify({'message': 'File too large'}), 413
+
+    success, message = fm.save_user_profile_image(user_id, file)
+    if success:
+        return jsonify({'message': message}), 200
+    else:
+        return jsonify({'message': message}), 400
+
+
+@APP.route('/fetch_profile_image', methods=["GET"])
+@verify_user
+def fetch_profile_image():
+    user_id = request.form['user_id']
+    user = User.query.get(user_id)
+    if user and user.profile_image_path:
+        # base_directory = mscolab_settings.UPLOAD_FOLDER
+        # base_path = os.path.join(base_directory, 'profile')
+        base_path = mscolab_settings.PROFILE_IMG_FOLDER
+        filename = os.path.basename(user.profile_image_path)
```

Comment:
The image path should be saved in a way in which it is relative to some configurable path (either `UPLOAD_FOLDER` or `PROFILE_IMG_FOLDER`). Using basename suggests that the saved path currently contains more information than it should (and using basename might throw away more than it should).

With my suggestions the saved path would be relative to `mscolab_settings.UPLOAD_FOLDER`. Relative to `PROFILE_IMG_FOLDER` would also be possible, that depends on if we want to make that configurable or not (again, see the other remark that it is currently configurable _and_ hardcoded in different places).

Reply:
@matrss The image path is actually saved relative to the `UPLOAD_FOLDER`
For example :` uploads/profile/1-20240627T210302-qqJvzG6QD3OrzBPAZKMkYbXArgI.jpg`
`os.path.basename(user.profile_image_path)` in this case is just used to get the file name `1-20240627T210302-qqJvzG6QD3OrzBPAZKMkYbXArgI.jpg`. I don't get whats wrong with this?

Reply:
That's not relative to UPLOAD_FOLDER, that is relative to the parent directory of UPLOAD_FOLDER. Taking the basename would fail if there was another directory in the path. If `user.profile_image_path` didn't start with uploads/ they could just be joined, without first using basename.